### PR TITLE
Selectable photos

### LINKED ISSUE #106: Possibilité de sélectionner les photos depuis la vue Photos
Actuellement dans la vue mobile, il n'y a pas moyen de sélectionner plusieurs images pour les partager par exemple - il serait top de prévoir un moyen de sélection avc checkbox et actions.
image

### PR DESCRIPTION
**Pull request by LunarX**
Closes #106

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt [hunk 1]
```diff
@@ -180,14 +180,16 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
         }
     }
 
-    fun deleteFileFromFavorites(file: File, userDrive: UserDrive? = null) = liveData(Dispatchers.IO) {
-        CoroutineScope(Dispatchers.IO).launch {
-            FileController.updateFile(file.id, userDrive = userDrive) {
-                it.isFavorite = false
+    fun deleteFileFromFavorites(file: File, userDrive: UserDrive? = null, callback: ((File) -> Unit)? = null) =
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt:183]: Rename `callback` into `onFileUpdate`

File: app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt [hunk 2]
```diff
@@ -168,25 +168,31 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
...
     }
 
-    fun addFileToFavorites(file: File, userDrive: UserDrive? = null, onSuccess: (() -> Unit)? = null) = liveData(Dispatchers.IO) {
-        ApiRepository.postFavoriteFile(file).let { apiResponse ->
-            if (apiResponse.isSuccess()) {
-                FileController.updateFile(file.id, userDrive = userDrive) { localFile ->
-                    localFile.isFavorite = true
+    fun addFileToFavorites(file: File, userDrive: UserDrive? = null, onSuccess: (() -> Unit)? = null) {
+        liveData(Dispatchers.IO) {
+            with(ApiRepository.deleteFavoriteFile(file)) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt:173]: Here we need to add to favorite, not delete

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt]: Fixed.

Comment (LunarX) [reply app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt]: The issue has been fixed

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/menu/PicturesAdapter.kt [hunk 3]
```diff
@@ -20,22 +20,33 @@ package com.infomaniak.drive.ui.menu
...
 import kotlinx.android.synthetic.main.cardview_picture.view.*
 import kotlinx.android.synthetic.main.title_recycler_section.view.*
+import java.util.*
 
-class PicturesAdapter(
-    private val onItemClick: (file: File) -> Unit
-) : LoaderAdapter<Any>() {
+class PicturesAdapter(private val onItemClick: (file: File) -> Unit) : LoaderAdapter<Any>() {
+
+    var itemsSelected: OrderedRealmCollection<File> = RealmList()
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/menu/PicturesAdapter.kt:40]: Rename to selectedItems

**Review by sirambd — APPROVED** [review]

**Pull request merged**